uPortal-security/uPortal-security-mvc/src/main/java/org/apereo/portal/url/ClassicMaxInactiveStrategy.java

```diff
@@ -0,0 +1,129 @@
+/*
+ Licensed to Apereo under one or more contributor license agreements. See the NOTICE file
+ distributed with this work for additional information regarding copyright ownership. Apereo
+ licenses this file to you under the Apache License, Version 2.0 (the "License"); you may not use
+ this file except in compliance with the License. You may obtain a copy of the License at the
+ following location:
+
+ <p>http://www.apache.org/licenses/LICENSE-2.0
+
+ <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ express or implied. See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+package org.apereo.portal.url;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apereo.portal.security.IAuthorizationPrincipal;
+import org.apereo.portal.security.IAuthorizationService;
+import org.apereo.portal.security.IPermission;
+import org.apereo.portal.security.IPerson;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component("maxInactiveStrategy")
+public class ClassicMaxInactiveStrategy implements IMaxInactiveStrategy {
+
+    public static final String MAX_INACTIVE_ATTR = "MAX_INACTIVE";
+
+    @Autowired private IAuthorizationService authorizationService;
+
+    @Override
+    public Integer calcMaxInactive(IPerson person) {
+        assert person != null;
+
+        IAuthorizationPrincipal principal =
+                authorizationService.newPrincipal(
+                        (String) person.getAttribute(IPerson.USERNAME), IPerson.class);
+        IPermission[] permissions =
+                authorizationService.getAllPermissionsForPrincipal(
+                        principal, IPermission.PORTAL_SYSTEM, MAX_INACTIVE_ATTR, null);
+
+        assert (permissions != null);
```

Comment:
Consider removing the parenthesis.   Noting line 35.

Reply:
:+1: